connector/failoverconnector/traces.go

```diff
@@ -28,11 +28,26 @@ func (f *tracesFailover) Capabilities() consumer.Capabilities {
 }
 
 // ConsumeTraces will try to export to the current set priority level and handle failover in the case of an error
-func (f *tracesFailover) ConsumeTraces(_ context.Context, _ ptrace.Traces) error {
-	return nil
+func (f *tracesFailover) ConsumeTraces(ctx context.Context, td ptrace.Traces) error {
+	for f.failover.pipelineIsValid() {
```

Comment:
I find it difficult to follow because we are generalizing two separate failure situations (currently stable pipeline fails vs looking for another stable pipeline), but then we still having to differentiate between these two cases later. I know it may not be quite as elegant, but what do you think about separating this into two cases for the sake of readability? 

First, try to send to the currently stable pipeline. Generally this will work and we can return immediately.

If that fails, fall into a loop to find the next stable pipeline.

Reply:
I see, yeah I think that would be reasonable. So something like

tc := f.failover.getCurrentConsumer()
err := tc.ConsumeTraces(ctx, td)
if err == nil {
    return
}
for f.failover.pipelineIsValid() { ......

Reply:
Yeah, basically. I think this would allow us to handle the two cases differently, rather than wait until later to re-determine if we need to.

Reply:
Wouldn't we know that if we have separated those cases?

Reply:
@djaglowski  I guess this also leads to the question of is there a different/better way to test if a pipeline is back to stable other than switching the pipeline that we're sending the core traffic to

Reply:
Actually yeah thats true as long as we still call reportStable if we returned no errors